### DISCUSSION TITLE
Strongly suggest sync_list usage in documentation

### DISCUSSION THREAD

**Discussion by russfink** [abraunegg/onedrive#2580] [Ideas]
My OneDrive has lots of data in it including videos, large image files, lots of things I do not need coming to 30Gb+, but I only need to sync one particular folder.  When I followed the instructions, onedrive attempted to sync the entire mess.  I could not find a command line option to onedrive to specify syncing only to a subfolder, but then saw the "sync_list" in the usage instructions.

Please mention this possible problem and to look at sync_list in the manpage and initial splash page.

This package is great btw!  Thank you!

**Comment by abraunegg** [comment]
@russfink 
Given that the documentation has been totally overhauled in v2.5.x, please review the following:
* https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-alpha-4/readme.md
* https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-alpha-4/docs/usage.md

> I could not find a command line option to onedrive to specify syncing only to a subfolder

Specifically in the v2.4.x documentation in the FAQ section there is the following:
* https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#how-to-sync-only-specific-or-single-directory

Utilise `--single-directory` option to only sync this specific path

**Comment by lauff** [comment]
I only need to sync one top-level directory, and wonder whether there is a disadvantage (e.g. performance, system resources) of using the sync_list with one line for the specific directory instead of the `--single-directory` directive.

**Reply by abraunegg** [reply]
@lauff 
In v2.5.x there is a programmatic difference between the two in terms of how the `--single-directory` path data online is collected, which (based on my testing) makes it take slightly longer to process and complete in a non-resync scenario as compared to using 'sync_list'. This is different to v2.4.x as in v2.5.x this feature was totally overhauled due to the change in handling shared folders.

In a resync scenario (which gets triggered when you modify or create 'sync_list') all objects need to be fetched from online due to the resync then evaluated against the 'sync_list' rules. This makes it a very time intensive process initially. Subsequent syncs however are much quicker as the data that needs to be synced is now cached.

In v2.4.x there is no programmatic difference when using `--single-directory`, however edge case errors were found with the logic which has been corrected in v2.5.x
